jit-service/src/main/java/org/dcsa/jit/service/notifications/EmailFormatter.java

```diff
@@ -0,0 +1,86 @@
+package org.dcsa.jit.service.notifications;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.dcsa.jit.persistence.entity.OperationsEvent;
+import org.dcsa.jit.persistence.entity.TimestampDefinition;
+import org.dcsa.jit.service.notifications.model.FormattedEmail;
+import org.dcsa.jit.service.notifications.model.MailConfiguration;
+import org.dcsa.jit.service.notifications.model.MailTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.regex.MatchResult;
+import java.util.regex.Pattern;
+
+@Component
+@RequiredArgsConstructor
+public class EmailFormatter {
+  private static final Pattern TEMPLATE_PATTERN = Pattern.compile("[{][{]\\s*([a-zA-Z-_./]+)\\s*[}][}]");
+
+  @Value("${dcsa.webui.baseUrl:NOT_SPECIFIED}")
+  private String webUIBaseUrl;
+
+  private final MailConfiguration mailConfiguration;
+
+  public FormattedEmail formatEmail(OperationsEvent operationsEvent, TimestampDefinition timestampDefinition, MailTemplate mailTemplate) {
+    Map<String, Function<OperationsEvent, Object>> customValues = Map.of(
+      "WEB_UI_BASE_URI", oe -> webUIBaseUrl,
+      "TRANSPORT_CALL_ID", oe -> oe.getTransportCall().getId(),
```

Comment:
This has changed to be the port visit ID. I think you can get it via:

`oe.getTransportCall().getPortVisit().getId()`

(the URL is unchanged - it would probably make sense to rename the subst variable as well)

Reply:
Added PORT_VISIT_ID to the list of substitutions. Not sure if we should remove TRANSPORT_CALL_ID

Reply:
Feel free to remove `TRANSPORT_CALL_ID` - we don't use it anymore (any use of it would be `PORT_VISIT_ID` instead)

Reply:
TRANSPORT_CALL_ID removed